File: test/javascript/list.js
var assert = require('assert');
var MinecraftVersions = require('minecraftverse');

describe('minecraft versions', function() {

  it('should get version manifest', function(done) {

    const apiClient = new MinecraftVersions.ApiClient('https://launchermeta.mojang.com');
    const api = new MinecraftVersions.DefaultApi(apiClient);

    var callback = function(error, data, response) {
      if (error) {
        done(error);
      } else {
        console.log('API called successfully. Returned data: ' + data);
        const versionManifest = data;
        const versions = versionManifest.versions;
        for (var i = 0; i < versions.length; i++) {
          const version = versions[i];
          console.log('version: ' + version.id);
          console.log('type: ' + version.type);
          console.log('url: ' + version.url);
        }
        done();
      }
    };
    api.getMinecraftVersionManifest(callback);
  });

  it('should get version package info', function(done) {

    const apiClient = new MinecraftVersions.ApiClient('https://piston-meta.mojang.com');
    const api = new MinecraftVersions.DefaultApi(apiClient);

    var callback = function(error, data, response) {
      if (error) {
        done(error);
      } else {
        console.log('API called successfully. Returned data: ' + data);
        const versionPackageInfo = data;
        console.log('version: ' + versionPackageInfo.id);
        assert.equal(typeof(versionPackageInfo.id), 'string')
        console.log('type: ' + versionPackageInfo.type);
        assert.equal(typeof(versionPackageInfo.type), 'string')
        const downloads = versionPackageInfo.downloads;
        for (var key in downloads) {
          console.log('download: ' + key);
          assert.equal(typeof(key), 'string')
          console.log('sha1: ' + downloads[key].sha1);
          assert.equal(typeof(downloads[key].sha1), 'string')
          console.log('size: ' + downloads[key].size);
          assert.equal(typeof(downloads[key].size), 'number')
          console.log('url: ' + downloads[key].url);
          assert.equal(typeof(downloads[key].url), 'string')
        }
        done();
      }
    };
    api.getMinecraftVersionPackageInfo('177e49d3233cb6eac42f0495c0a48e719870c2ae', '1.21', callback);
  });

});

File: test/python/list.py
import unittest
import minecraftverse
from minecraftverse.rest import ApiException
from pprint import pprint

class TestMinecraftVersion(unittest.TestCase):

    def test_get_version_manifest(self):

      configuration = minecraftverse.Configuration(
          host = "https://launchermeta.mojang.com"
      )

      with minecraftverse.ApiClient(configuration) as api_client:

          api_instance = minecraftverse.DefaultApi(api_client)

          try:
              api_response = api_instance.get_minecraft_version_manifest()
              print("The response of DefaultApi->get_minecraft_version_manifest:\n")
              # pprint(api_response)
              version_manifest = api_response;
              print(version_manifest.versions)
              for version in version_manifest.versions:
                  print('version: ' + version.id)
                  assert isinstance(version.id, str)
                  print('type: ' + version.type)
                  assert isinstance(version.type, str)
                  print('url: ' + version.url)
                  assert isinstance(version.url, str)

          except ApiException as e:
              self.fail('Exception when calling DefaultApi->get_minecraft_version_manifest: %s\n' % e)

    def test_get_version_package_info(self):

      configuration = minecraftverse.Configuration(
          host = "https://piston-meta.mojang.com"
      )

      with minecraftverse.ApiClient(configuration) as api_client:

          api_instance = minecraftverse.DefaultApi(api_client)

          try:
              api_response = api_instance.get_minecraft_version_package_info('177e49d3233cb6eac42f0495c0a48e719870c2ae', '1.21')
              print("The response of DefaultApi->get_minecraft_version_manifest:\n")
              pprint(api_response)
              version_package_info = api_response
              print('version: ' + version_package_info.id)
              assert isinstance(version_package_info.id, str)
              print('type: ' + version_package_info.type)
              assert isinstance(version_package_info.type, str)
              version_package_info_downloads = version_package_info.downloads
              server_download = version_package_info_downloads.server
              print('sha1: ' + server_download.sha1)
              assert isinstance(server_download.sha1, str)
              print('size: ' + str(server_download.size))
              assert isinstance(server_download.size, int)
              print('url: ' + server_download.url)
              assert isinstance(server_download.url, str)
          except ApiException as e:
              self.fail('Exception when calling DefaultApi->get_minecraft_version_manifest: %s\n' % e)
